src/lowering/desugar.ml

```diff
@@ -334,20 +334,39 @@ and export_interface txt =
   let open T in
   let name = "__get_candid_interface_tmp_hack" in
   let v = "$__get_candid_interface_tmp_hack"  in
-  let binds = [T.scope_bind] in
+  let binds = [scope_bind] in
   let typ = Func (Shared Query, Promises, binds, [], [text]) in
 
-  let scope_con = Cons.fresh "T" (Abs ([], T.scope_bound)) in
+  let scope_con = Cons.fresh "T" (Abs ([], scope_bound)) in
   let scope_con2 = Cons.fresh "T2" (Abs ([], Any)) in
-  let bind  = typ_arg scope_con T.Scope T.scope_bound in
-  let bind2 = typ_arg scope_con2 T.Scope T.scope_bound in
+  let bind  = typ_arg scope_con Scope scope_bound in
+  let bind2 = typ_arg scope_con2 Scope scope_bound in
   ([ letD (var v typ) (
     funcE v (Shared Query) Promises [bind] [] [text] (
-      asyncE bind2 (textE txt) (T.Con (scope_con, []))
+      asyncE bind2 (textE txt) (Con (scope_con, []))
     )
   )],
   [{ it = { I.name = name; var = v }; at = no_region; note = typ }])
 
+and export_footprint self_id name expr =
+  let open T in
+  let v = "$__stable_variable_footprint" in
+  let binds = [scope_bind] in
+  let typ = Func (Shared Query, Promises, binds, [], [nat64]) in
+
+  let scope_con = Cons.fresh "T" (Abs ([], scope_bound)) in
+  let scope_con2 = Cons.fresh "T2" (Abs ([], Any)) in
+  let bind  = typ_arg scope_con Scope scope_bound in
+  let bind2 = typ_arg scope_con2 Scope scope_bound in
+  ([ letD (var v typ) (
+       funcE v (Shared Query) Promises [bind] [] [nat64] (
+           (asyncE bind2
+              (blockE [expD (assertE (primE (I.RelPrim (caller, Operator.EqOp))
```

Comment:
The assertion message produced below is a bit obscure - maybe call OtherPrim "trap" with `"__motoko_stable_var_size: not a self-call"` or similar to what `__motoko_async_helper` produces (see compile.ml)

Reply:
Hmm, I am reluctant to add user-friendliness for things that are clearly off-limits to users. It only adds weight, but no value. Who calls (manages to call) this will be able to disassemble it (or find the generating code).